clang/lib/AST/Stmt.cpp

```diff
@@ -1482,3 +1482,10 @@ bool CapturedStmt::capturesVariable(const VarDecl *Var) const {
 
   return false;
 }
+
+Stmt *LoopControlStmt::getLabelTarget() const {
+  Stmt *Target = TargetLabel->getStmt();
+  while (isa_and_present<LabelStmt>(Target))
+    Target = cast<LabelStmt>(Target)->getSubStmt();
+  return Target;
```

Comment:
The name similarity between `TargetLabel` (which stores the label) and `LabelTarget` is pretty confusing/awkward.  It seems that this is attempting to get the control statement kind that this refers to, right?  So the `for`/`while`/`switch`/etc?

I wonder if INSTEAD this should JUST return the `TargetLabel`, and add a function to `LabelStmt` that is `getInnermostLabeledStmt` or something (choose a better name).  This way it is more clear what is actually going on here (this is/was quite confusing for me so far).

Reply:
I’ve added `getInnermostLabeledStmt()` because it seems like a useful thing to have, and I’ve renamed this function to `getNamedLoopOrSwitch()` to avoid having to write `getLabelDecl()->getStmt()->getInnermostLabeledStmt()` in a number of places.